src/main/kotlin/org/malachite/estella/services/RecruitmentProcessService.kt

```diff
@@ -100,9 +101,23 @@ class RecruitmentProcessService(
     fun updateEndDate(processId: Int, endDate: Date) {
         val userFromJWT = securityService.getHrPartnerFromContext()
         val process = getProcess(processId)
-        if (process.offer.creator.id != userFromJWT?.id) throw UnauthenticatedException()
-        if (process.startDate.after(endDate)) throw InvalidEndDateException()
+        assertCanPerformOperation(process, userFromJWT)
+        if (process.isStarted()) throw ProcessAlreadyStartedException(processId)
+        if (process.startDate == null && Date.from(Instant.now()).after(endDate)) throw InvalidEndDateException()
+        if (process.startDate?.after(endDate) == true) throw InvalidEndDateException()
         recruitmentProcessRepository.save(process.copy(endDate = endDate))
     }
 
+    fun startProcess(processId: Int) {
+        val user = securityService.getHrPartnerFromContext()
+        val process = getProcess(processId)
+        if (process.startDate != null) throw ProcessAlreadyStartedException(processId)
+        assertCanPerformOperation(process, user)
```

Comment:
To pierdoła ale nie wiem czy nie powinniśmy najpierw sprawdzić czy użytkownik ma prawo coś zrobić a dopiero potem sprawdzać czy proces się zaczął